draft-ietf-6man-slaac-renum-08.xml

```diff
@@ -379,94 +488,143 @@ The behavior described in <xref target="RFC4862"/> had been incorporated during
+</t>
 
-   In such cases, the router MAY transmit up to
-   MAX_INITIAL_RTR_ADVERTISEMENTS unsolicited advertisements, using the
-   same rules as when an interface becomes an advertising interface.
+
+<t>
+To enable the proposed mechanism, this document introduces the following changes to Section 6.2.3 of <xref target="RFC4861"/>:
 </t>
-</list>
 
-with:
+<t>
+OLD TEXT:
+</t>
+<t>
+===
+</t>
+<t>
+A router MAY choose not to include some or all options when sending unsolicited Router Advertisements. For example, if prefix lifetimes are much longer than AdvDefaultLifetime, including them every few advertisements may be sufficient. However, when responding to a Router Solicitation or while sending the first few initial unsolicited advertisements, a router SHOULD include all options so that all information (e.g., prefixes) is propagated quickly during system initialization.
+</t>
+<t>
+If including all options causes the size of an advertisement to exceed the link MTU, multiple advertisements can be sent, each containing a subset of the options.
+</t>
+<t>
+===
+</t>
 
-<list style="hanging">
 <t>
-   In such cases, the router SHOULD transmit 
-   MAX_INITIAL_RTR_ADVERTISEMENTS unsolicited advertisements, using the
-   same rules as when an interface becomes an advertising interface.
+NEW TEXT:
 </t>
-</list>
+<t>
+===
+</t>
+<t>
+When sending Router Advertisements, a router SHOULD include all options into each RA. However if prefix-specific link-local addresses [draft-ietf-6man-slaac-renum] are enabled for the interface, the router SHOULD include one PIO into each RA, as described in [draft-ietf-6man-slaac-renum].
+If including all required options would cause the size of an advertisement to exceed the link MTU, multiple advertisements can be sent, each containing a subset of the options. 
+</t>
+<t>
+===
+</t>
+
+<t>
+This document also modifies Section 6.2.8 of <xref target="RFC4861"/>:
+</t>
+<t>
+===
+</t>
+<t>OLD TEXT:</t>
+<t>
+===
+</t>
+<t>
+Using the link-local address to uniquely identify routers on the link has the benefit that the address a router is known by should not change when a site renumbers.
 </t>
+<t>
+===
+</t>
+<t>NEW TEXT:</t>
+<t>
+===
+</t>
+<t>
+Using the link-local address to uniquely identify routers on the link has the benefit that the address a router is known by should not change when a site renumbers and the renumbering event is explicitly signalled and properly propagated to all hosts.
+However, in case of flash renumbering without explicit signalling changing the router link-local address during the renumbering ([draft-ietf-6man-slaac-renum]) helps hosts to detect prefix changes and update their configuration accordingly.
```

Comment:
I feel like we should use normative language here.

Reply:
I agree with the normative language suggestion. However, re: DHCPv6-PD vs non-PD cases...

Pls note the text above in the same section, about per-prefix functionality:

"That functionality SHOULD be configurable. A routers supporting this functionality SHOULD enable it on an interface if AdvPrefixList contains any prefixes obtained via DHCPv6-PD. In all other cases this functionality SHOULD be disabled by default unless explictly enabled by the administrator."

Think about cases like a large corporate network or a datacenter. My MX960s etc boxes have prefixes (and sometimes even link-local addresses) statically configured. If I (or my automation system) changes a prefix on the router interface (e.g. vlan), it's not going to be a flash-renumbering event, as my automation can ensure that old prefixes are properly deprecated (or even can be used in parallel with the new ones). So it would be completely unnecessary to complicate things and make new LLAs in such scenario. What we really care about in flash-renumbering case is routers where prefix configuration on the interface can change dynamically.

What do you think?

Reply:
I've updated the text slightly, leaving the door open for routers/scenarios when it is not needed.